subworkflows/local/bam_variant_calling_somatic_mutect2/main.nf

```diff
@@ -36,8 +37,23 @@ workflow BAM_VARIANT_CALLING_SOMATIC_MUTECT2 {
         // Move num_intervals to meta map and reorganize channel for MUTECT2_PAIRED module
         .map{ meta, input_list, input_index_list, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], input_list, input_index_list, intervals ] }
 
-    // Perform variant calling using mutect2 module pair mode
-    MUTECT2_PAIRED( input_intervals, fasta, fai, dict, germline_resource, germline_resource_tbi, panel_of_normals, panel_of_normals_tbi)
+    if (joint_mutect2) {
+        // Separate normal cram files and remove duplicates
+        ch_normal_cram = input.map{ meta, cram, crai -> [ meta - meta.subMap('tumor_id') + [id:meta.patient], cram[0], crai[0] ] }.unique()
+        // Extract tumor cram files
+        ch_tumor_cram = input.map{ meta, cram, crai -> [ meta - meta.subMap('tumor_id') + [id:meta.patient], cram[1], crai[1] ] }
+        // Merge normal and tumor crams by patient
+        ch_tn_cram = ch_normal_cram.mix(ch_tumor_cram).groupTuple()
```

Comment:
oh no a groupTuple without a size. This will stall the workflow and cause a bottleneck. I don't have a good idea at the moment on how to properly retrieve the size. Can you open an issue please, that we need to figure this out and fix at some point?

Reply:
https://github.com/nf-core/sarek/issues/1141